### DISCUSSION TITLE
K3s upgrade single node cluster

### DISCUSSION THREAD

**Discussion by Wuppie007** [mysticaltech/terraform-hcloud-kube-hetzner#1092] [Q&A]
I'm running a single node cluster for a couple of months now, and I was planning to update de k3s version. 

I've found [k3s upgrades](https://docs.k3s.io/upgrades/manual) to do it manually but i'm nog sure about the existing arguments (EXISTING_K3S_ENV).

I also ran into "Automatic upgrades", also mentioned in this [discussion](https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/discussions/381). Is this also working for single node clusters?

**Comment by MariusSchiffer** [comment]
I'm usually adding the automatic upgrade plan with `cordon: false` to single-node clusters, but setting the `k3s_upgrade=false` label to the node. When I want to upgrade, I set the label to true and restart the upgrade controller deployment. Then I can monitor the upgrade.

Afterwards I set the label back to `k3s_upgrade=false`.
